chart/templates/secret.yaml

```
@@ -0,0 +1,11 @@
+# Comment out this secret if you want to use hardcoded password from {{ .Values.env.rds.password.hardcoded }}
+{{- if (default false .Values.env.rds.override) }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: rds-secret  # This must match `{{ .Values.env.rds.password.secret }}`
+  namespace: {{ .Values.namespace }}
+type: Opaque
+data:
+  password: cGFzc3dvcmQ=  # Base64-encoded value of the password, can fallback from `{{ .Values.env.rds.password.hardcoded }}`
+{{- end }}
```

```suggestion
{{- if and (default false .Values.env.rds.override) (not (lookup "v1" "Secret" .Release.Namespace .Values.env.rds.password.secret)) }}
apiVersion: v1
kind: Secret
metadata:
  name: {{ .Values.env.rds.password.secret }}
  namespace: {{ .Release.Namespace }}
  annotations:
    "helm.sh/hook": "pre-install"
    "helm.sh/hook-delete-policy": "before-hook-creation"
type: Opaque
data:
  password: {{ .Values.env.rds.password.secret.hardcoded | b64enc }}
{{- end }}
```
I think lookup check belongs in secret. The deployment should not do helm lookup instead secret should, the secret should always be there wrt the deployment as long as override is true.

---

Please check the helm syntax for and condition and others, I may have made a mistake. But the basic idea is this only.

---

The syntax is correct :)

But is there any reason why this should be created part of pre-install hooks?

---

* helm.sh/hook: pre-install <-- Runs the resource only on install.
* helm.sh/hook-delete-policy: before-hook-creation <-- Deletes any previous failed pre-install hook runs before re-creating it.

---

👍 Added it - https://github.com/waii-ai/waii-helm-chart/pull/6/commits/a5cedbef2da211ab7a5102304d5518782b02a251